VisualPinball.Unity/VisualPinball.Unity.Editor/VPT/Sounds/SoundInspector.cs

```diff
@@ -0,0 +1,357 @@
+				_clipIndex++;
+			}
+
+			else
+			{
+				_clipIndex = 0;
+				PlayClip(_clipArray[_clipIndex]);
+				_clipIndex++;
+			}
+		}
+
+
+
+		private void PlayRandom(AudioClip[] _clipArray)
+		{
+			if (_clipIndex == _clipArray.Length)
+			{
+				_clipIndex = 0;
+			}
+
+			if (_clipArray.Length > 1)//randomize clip to play only if more than one clip
+			{
+				_clipIndex = Random.Range(0, _clipArray.Length);
+				PlayClip(_clipArray[_clipIndex]);
+			}
+			else 
+			{ PlayClip(_clipArray[_clipIndex]); }
+
+		}
+
+		
+		private void PlayClip(AudioClip clip)
+		{
+			GameObject tempGameObject = new GameObject("Sound");
+			tempGameObject = GetTempGameObject(tempGameObject);
+			AudioSource audioSource = tempGameObject.GetComponent<AudioSource>();
+			audioSource.clip = clip;
+			_clipPlaying = true;
+			audioSource.Play();
+			
+
+		}
+		private  GameObject[] GetSoundObjects()
+		{
+			GameObject[] objects = new GameObject[_clipArray.Length];
+
+			for (int i = 0; i < _clipArray.Length; i++)
+			{
+				GameObject tempGameObject = new GameObject($"Sound{i}");
+				tempGameObject = GetTempGameObject(tempGameObject);
+				AudioSource audioSource = tempGameObject.GetComponent<AudioSource>();
+				audioSource.clip = _clipArray[i];
+				objects[i] = tempGameObject;
+			}
+
+			_soundObjectsLength = objects.Length;
+
+			return objects;
+			
+
+		}
+
+		// Create a temporary gameobject so that a temporary audiosource can be attached to it, in order to play audioclips
+		private GameObject GetTempGameObject(GameObject tempGameObject)
+		{
+
+			Vector3 position = new Vector3(5, 1, 2);
+			float volume = VolumeCorrection.floatValue;
+			float pitchModifier = RandomizePitch.floatValue;
+			float speedModifier = RandomizeSpeed.floatValue;
+			float volumeModifier = RandomizeVolume.floatValue;
+			
+			if (volumeModifier != 1)
+			{ volume = volumeModifier; }
+
+			tempGameObject.transform.position = position;
+			//_tempGameObject.hideFlags = HideFlags.HideAndDontSave; //dont save object in editor and dont show in heirarchy
+			AudioSource tempAudioSource = (AudioSource)tempGameObject.AddComponent(typeof(AudioSource));
+			tempAudioSource.spatialBlend = 1f;
+			tempAudioSource.pitch =  pitchModifier;
+			tempAudioSource.volume = volume;
+			float value;
+
+			AudioMixer audioMixer = Resources.Load<AudioMixer>("SoundMixer");
+			AudioMixerGroup[] audioMixGroup = audioMixer.FindMatchingGroups("Master");
+			audioMixGroup[0].audioMixer.SetFloat("pitchShifter", speedModifier);
+			//audioMixGroup[0].audioMixer.GetFloat("pitchShifter", out value);
+			//Debug.Log(value.ToString());
+			tempAudioSource.outputAudioMixerGroup = audioMixGroup[0];
+
+			return tempGameObject;
+
+		}
+		
+		private void Update()
+		{
+			
+			if (_loop)
+			{
+				//set up new array of gameobjects based on most current number of audioclips, if new loop is initiated
+				if (_soundObjects == null || _soundObjectsLength == 0) 
+				{ _soundObjects = GetSoundObjects();}
```

Comment:
What's the reason for this? Is it impossible to schedule it on the same audio source?

I'll need to dig more into Unity's audio system, but IMO you should have an audio source on each GameObject that emits audio, and play the also the editor sound through that. (Adding the audio source automatically can be easily achieved with `[RequireComponent(typeof(AudioSource))]` on `MechSoundsComponent`.)